e2e_testing/xfail_sets.py

```diff
@@ -24,6 +24,8 @@
 
 MHLO_PASS_SET = {
     "BroadcastToIdentityCaseStaticModule_basic",
+    "GatherStaticModule_basic",
+    "GatherModule_basic",
```

Comment:
Could you please add more dynamic shapes e2e tests to cover the following cases:
1. Both input and index are randomly generated; (The index could be un-ordered)
2. Test data with different ranks

Reply:
Done.